### ISSUE TITLE
Bug: indentation problem with return statement in conditional

### ISSUE THREAD

**Issue by PierBover** [AtomLinter/linter-eslint#807]
I'm reposting [this issue](https://github.com/eslint/eslint/issues/8046) here since the error does not reproduce when using http://eslint.org/demo/.

For some reason this doesn't throw an indentation error:
```js
if (items[i].id == id) {
return items[i];
}
```
But this (obviously) does although only in the `foo` line:
```js
if (items[i].id == id) {
var foo = 'bar';
return items[i];
}
```

![image](https://cloud.githubusercontent.com/assets/226483/22754275/a6d6172c-ee04-11e6-9926-7809d785ed7d.png)

I would expect the `return` line to throw an error since it's not indented.

I'm using this rule for indents:
```json
"indent": ["error", "tab", {"SwitchCase": 1}]
```

Here is the complete set of rules: https://gist.github.com/PierBover/992647a6fe163dfadf1cea3243c96c68

```
Atom version: 1.13.1
linter-eslint version: 8.1.0
ESLint version: 3.14.1
Hours since last Atom restart: 0.9
Platform: darwin
Using bundled fallback ESLint from /Users/pier/.atom/packages/linter-eslint/node_modules/eslint
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": []
}
```

**Comment by Arcanemagus** [comment]
I'm also unable to reproduce this:
![image](https://cloud.githubusercontent.com/assets/427137/22839445/5063cf3a-ef7e-11e6-9d38-fb6829fa7ace.png)

I used the following files to test this: [linter-eslint_GH807.zip](https://github.com/AtomLinter/linter-eslint/files/767528/linter-eslint_GH807.zip)

Are you seeing this with the above files? If not, you will have to provide files that you are seeing this on.

**Comment by Arcanemagus** [comment]
Closing due to a lack of response and the inability to reproduce the problem.
